packages/daemon/src/daemon-node-powers.js

```diff
@@ -279,111 +375,208 @@ export const makePowers = ({
     }
   };
 
-  /**
-   * @param {string} path
-   */
-  const makeFileReader = path => {
-    const nodeReadStream = fs.createReadStream(path);
-    return makeNodeReader(nodeReadStream);
+  const makeHashedContentReadeableBlob = (statePath, sha512) => {
+    const storageDirectoryPath = diskPowers.joinPath(
+      statePath,
+      'store-sha512',
+    );
+    const storagePath = diskPowers.joinPath(storageDirectoryPath, sha512);
+    const stream = async () => {
+      const reader = diskPowers.makeFileReader(storagePath);
+      return makeReaderRef(reader);
+    };
+    const text = async () => {
+      return diskPowers.readFileText(storagePath);
+    };
+    const json = async () => {
+      return JSON.parse(await text());
+    };
+    return { stream, text, json }
   };
 
-  /**
-   * @param {string} path
-   */
-  const makeFileWriter = path => {
-    const nodeWriteStream = fs.createWriteStream(path);
-    return makeNodeWriter(nodeWriteStream);
-  };
+  const makeHashedContentWriter = async (statePath) => {
```

Comment:
naming is hard

Reply:
Yeah, how about `makeContentHashWriter`? `makeContentAddressWriter`? Maybe step back to `makeContentAddressStore` with `store` and `fetch` instead?